libnavui-util/src/main/java/com/mapbox/navigation/ui/utils/resource/ResourceLoaderFactory.kt

```diff
@@ -0,0 +1,18 @@
+package com.mapbox.navigation.ui.utils.resource
+
+import com.mapbox.common.TileStore
+
+/**
+ * Factory for accessing ResourceLoader instance.
+ */
+object ResourceLoaderFactory {
+
+    private val sharedLoader: DefaultResourceLoader by lazy {
+        DefaultResourceLoader(TileStore.create())
```

Comment:
How do you see developers passing a custom storage path here? This could potentially come as a requirement down the line. What if the APIs that use the resource loader take a TileStore as optional argument?

Reply:
Catching our slack conversation here.
We'll add a new `load` method that will allow TileStore injection.

```kotlin
abstract fun load(tileStore: TileStore, request: ResourceLoadRequest, callback: ResourceLoadCallback): Long
```